src/pages/rayon.md

```diff
@@ -0,0 +1,46 @@
+# Rayon
+[![rayon-badge]][rayon]
+
+## Calculate difference of value 1 from each element of an array
+
+```rust
+
+/**
+ * An example of parallel processing by using rayon where difference of
+ * 1 is calculated on each element of array in parallel
+ */
+
+extern crate rayon;
+use rayon::prelude::*;
+
+//function to calculate difference of 1 one on each element of an array
+fn difference(input: &mut[i32]){
+    // par_iter_mut() uses internal thread pool to easily parallelize data-parallel operations
+    input.par_iter_mut().for_each(|p| *p -= 1);
+    //print the output,i.e.the new mutated elements of array
+    println!("The mutated array elements are:");
+    for i in input.iter() {
+        println!("{}", *i)
+    };
+}
+
+fn main() {
+    let arr = &mut [0, 7, 9, 11];
+    difference(arr);
+}
+```
```

Comment:
Let's condense this a bunch! Since we're creating a book, general commentary about rayon can be done in the book text, and much of the remaining comments are redundant with the code, or the text, obscuring the mechanisms on display.

Really, if you think about what's 'real' code here it boils down to:

```
fn main() {
    let arr = &mut [0, 7, 9, 11];

    arr.par_iter_mut().for_each(|p| *p -= 1);

    for i in arr.iter() {
        println!("{}", *i)
    }
}
```

As an example, that looks more crisp to me. But I think we can make this even more succinct. Looping through that array to print stuff takes a lot of code, but it's kinda beside the point of the example. Instead of printing each element let's just dump the entire array at once:

```
fn main() {
    let arr = &mut [0, 7, 9, 11];

    arr.par_iter_mut().for_each(|p| *p -= 1);

   println!("{:?}", arr);
}
```

Real short and sweet. It's obvious that `par_iter_mut` is the thing we're showing off. This is a simple example users can copy into their own code.